sdk/src/types/block/output/feature/metadata.rs

```diff
@@ -35,72 +35,29 @@ type MetadataBTreeMap =
 
 /// Defines metadata, arbitrary binary data, that will be stored in the output.
 #[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
-pub struct MetadataFeature(
-    // Binary data.
-    pub(crate) MetadataBTreeMapPrefix,
-);
+pub struct MetadataFeature(MetadataBTreeMapPrefix);
 
-impl Packable for MetadataFeature {
-    type UnpackError = Error;
-    type UnpackVisitor = ();
-
-    fn pack<P: Packer>(&self, packer: &mut P) -> Result<(), P::Error> {
-        self.0.pack(packer)?;
-
-        Ok(())
-    }
-
-    fn unpack<U: Unpacker, const VERIFY: bool>(
-        unpacker: &mut U,
-        visitor: &Self::UnpackVisitor,
-    ) -> Result<Self, UnpackError<Self::UnpackError, U::Error>> {
-        let mut unpacker = CounterUnpacker::new(unpacker);
-        let start_opt = unpacker.read_bytes();
-
-        let map = MetadataBTreeMapPrefix::unpack::<_, VERIFY>(&mut unpacker, visitor)
-            .map_packable_err(|e| Error::InvalidMetadataFeature(e.to_string()))?;
-
-        verify_keys_packable::<true>(&map).map_err(UnpackError::Packable)?;
-
-        let packed_len = if let (Some(start), Some(end)) = (start_opt, unpacker.read_bytes()) {
-            end - start
-        } else {
-            map.packed_len()
-        };
-
-        let packed_len = u16::try_from(packed_len).map_err(|_| {
-            UnpackError::Packable(Error::InvalidMetadataFeature(format!(
-                "Out of bounds byte length: {}",
-                packed_len + 1
-            )))
-        })? + 1; // +1 for the type byte
-
-        if !Self::LENGTH_RANGE.contains(&(packed_len)) {
-            return Err(UnpackError::Packable(Error::InvalidMetadataFeature(format!(
-                "Out of bounds byte length: {packed_len}"
-            ))));
-        }
-
-        Ok(Self(map))
-    }
+fn verify_packable<const VERIFY: bool>(feature: &MetadataFeature) -> Result<(), Error> {
+    verify_keys_packable::<VERIFY>(feature)?;
+    verify_byte_length_packable::<VERIFY>(feature.packed_len())?;
+    Ok(())
 }
 
-fn verify_keys_packable<const VERIFY: bool>(map: &MetadataBTreeMapPrefix) -> Result<(), Error> {
+fn verify_keys_packable<const VERIFY: bool>(feature: &MetadataFeature) -> Result<(), Error> {
     if VERIFY {
-        for key in map.keys() {
-            if !key.iter().all(|b| b.is_ascii_graphic()) {
+        for key in feature.0.keys() {
+            if !key.iter().all(|c| c.is_ascii_graphic()) {
                 return Err(Error::NonGraphicAsciiMetadataKey(key.to_vec()));
             }
         }
     }
     Ok(())
 }
 
-fn verify_length_packable<const VERIFY: bool>(map: &MetadataBTreeMapPrefix) -> Result<(), Error> {
+fn verify_byte_length_packable<const VERIFY: bool>(len: usize) -> Result<(), Error> {
     if VERIFY {
         // +1 for the feature type
```

Comment:
What happened to this?

Reply:
Yes but the +1 is gone?

Reply:
because I'm packing the whole feature

Reply:
But the type is in the feature enum, not in the feature itself

Reply:
Actually checked the TIP again and the limit is without the type, so just removing the comment